### PR TITLE
Change date based on date format set in settings

### LINKED ISSUE #3873: Table view / Dates (3 pts)
In table view, "Date added" property is shown in Month-Day-Year format, and not as configured in Collection.

And "Date modified" is shown in the internal format, like "1629158545556".

sc

### PR DESCRIPTION
**Pull request by grafitto**
fixes #3873 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Comment by grafitto** [comment]
image
Should "Date added" also conform to date format added on the settings? @konzz @RafaPolit

File: app/react/Library/components/TableCell.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { I18NLink } from 'app/I18N';
 import GeolocationViewer from 'app/Metadata/components/GeolocationViewer';
 import { LinkSchema, MetadataObjectSchema, PropertySchema } from 'shared/types/commonTypes';
 import MarkdownViewer from 'app/Markdown';
+// import moment from 'moment';
```
Comment (LaszloKecskes) [inline app/react/Library/components/TableCell.tsx:7]: Please remove commented imports, if we are sure that they are not used.

File: app/react/Library/components/TableRow.tsx [hunk 2]
```diff
@@ -81,12 +81,13 @@ class TableRowComponent extends Component<TableRowProps> {
...
     const { entity, templates, thesauris, columns, selected } = this.props;
     const formattedEntity = formatter.prepareMetadata(entity.toJS(), templates, thesauris, null, {
-      sortedProperty: 'creationDate',
+      sortedProperties: ['editDate', 'creationDate'],
     });
     const columnValues = new Map();
     formattedEntity.metadata.forEach((prop: FormattedMetadataValue) => {
       columnValues.set(prop.name, prop);
     });
+    console.log('columns: ', columns);
```
Comment (LaszloKecskes) [inline app/react/Library/components/TableRow.tsx:90]: Also remove this log.

File: app/react/Library/components/TableCell.tsx (hunk 1 above)
Comment (grafitto) [reply app/react/Library/components/TableCell.tsx]: Removed

File: app/react/Library/components/TableRow.tsx (hunk 2 above)
Comment (grafitto) [reply app/react/Library/components/TableRow.tsx]: removed

File: app/react/Metadata/helpers/formater.js [hunk 3]
```diff
@@ -347,7 +347,7 @@ export default {
     });
   },
 
-  prepareMetadata(_doc, templates, thesauris, relationships, options = {}) {
+  prepareMetadata(_doc, templates, thesauris, relationships, options = { sortedProperties: [] }) {
```
Comment (LaszloKecskes) [inline app/react/Metadata/helpers/formater.js]: I think this a bit different than the previous behavior. Now, what if I pass an options object, but it doesn't have sortedProperties? The previous solution supplies that to any options object that's missing it, this one only if the entire options object is missing.

Comment (grafitto) [reply app/react/Metadata/helpers/formater.js]: I have changed this

**Review by LaszloKecskes — APPROVED** [review]
After the discussed changes, looks good.

**Pull request merged**